README.md

```diff
@@ -48,33 +65,470 @@ cache read/white process, etc. Command handler can be sync and publish the
+
+```swift
+Engine.sharedInstance.invalidate()
+```
+
+## Life hacks
+
+### Stories
+
+Naming is hard. I doesn't seem right to have names like `BookListCommand`,
+`BookListCommandHandler` and `BookListWhatever` in the project. Here comes the
+idea of stories that group related types and make the flow more concrete:
+
+```swift
+struct BookListStory {
+
+  struct Command: Aftermath.Command {
+    // ...
+  }
+
+  struct Handler: Aftermath.CommandHandler {
+    // ...
+  }
+}
+```
+
+In this sense it's close to user stories used in agile software development
+methodologies.
+
+You can find more detailed example in [AftermathNotes](https://github.com/hyperoslo/Aftermath/blob/master/Example/Aftermath)
+demo project.
+
+### Features
+
+Some of the stories may seem very similar. Then in makes sense to make them
+more generic and reusable according to the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
+principle. For example, let's say we have the flow to fetch a single resource
+by id.
+
+```swift
+import Aftermath
+import Malibu
+
+// Generic feature
+protocol DetailFeature {
+  associatedtype Model: Entity
+  var resource: String { get }
+}
+
+// Command
+struct DetailCommand<Feature: DetailFeature>: Aftermath.Command {
+  typealias Output = Feature.Model
+  let id: Int
+}
+
+// Command handler
+struct DetailCommandHandler<Feature: DetailFeature>: Aftermath.CommandHandler {
+  typealias Command = DetailCommand<Feature>
+
+  let feature: Feature
+
+  func handle(command: Command) throws -> Event<Command> {
+    fetchDetail("\(feature.resource)/\(command.id)") { json, error in
+      if let error = error {
+        self.publish(error: error)
+        return
+      }
+
+      do {
+        self.publish(data: try Feature.Model(json))
+      } catch {
+        self.publish(error: error)
+      }
+    }
+
+    return Event.Progress
+  }
+}
+
+// Concrete feature
+struct BookFeature: ListFeature, DeleteFeature, CommandProducer {
+  typealias Model = Todo
+  var resource = "books"
+}
+
+// Execute command to load a single resource.
+execute(command: DetailCommand<BookFeature>(id: id))
+
+// Register reaction listener.
+react(to: DetailCommand<BookFeature>.self, with: reaction)
+```
+
+You can find more detailed example in [AftermathNotesPlus](https://github.com/hyperoslo/Aftermath/blob/master/Example/AftermathNotes)
+demo project.
+
+## Summary
+
+We believe that in iOS applications in most of the cases there is no real need
+for single global state (single source of truth) or multiple sub-states
+distributed between stores. Data is stored on disc in local persistence layer,
+such as database and cache, or it's fetched from network. Then this content,
+assembled piece by piece from different sources, is translated into the
+"view state", which is readable by the view to render it on the screen. This
+"view state" is kept in memory and valid at a given instant in time until we
+switch the context and the current view is deallocated.
```

Comment:
From:
> We believe that in iOS applications in most of the cases there is no real need
for single global state (single source of truth) or multiple sub-states
distributed between stores. Data is stored on disc in local persistence layer,
such as database and cache, or it's fetched from network. Then this content,
assembled piece by piece from different sources, is translated into the
"view state", which is readable by the view to render it on the screen. This
"view state" is kept in memory and valid at a given instant in time until we
switch the context and the current view is deallocated.

To:
> We believe that in iOS applications, in most of the cases, there is no real need
for single global state (single source of truth) or multiple sub-states
distributed between stores. Data is stored on disc in local persistence layer,
such as database and cache, or it's fetched from network. Then this content,
assembled piece by piece from different sources, is translated into the
"view state", which is readable by the view to render it on the screen. This
"view state" is kept in memory and valid at a given instant in time until we
switch the context and the current view is deallocated.